docs/customization/api.rst

```diff
@@ -95,3 +95,198 @@ The next step is to modify the security configuration in ``config/packages/secur
     Changing prefix without security configuration update can expose confidential data (like customers addresses).
 
 After these two steps you can start to use endpoints with new prefixes
+
+How to customize serialization?
+===============================
+
+Let's assume that you want to modify responses with your custom fields serialized in response.
+For an example we will use ``Product`` resource and customize its fields.
+
+Adding a field to response
+==========================
+
+Let's say that you want to add a new field named ``additionalText`` to ``Product``.
+First let's create a new serializer that will supports our ``Product`` resource.
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Serializer;
```

Comment:
I am not sure about namespace here and in xml service declaration. Btw is xml ok here or should I make a yaml counterpart?

Reply:
The namespace is OK for me, but I would be for adding the service definition in YAML as it is the standard in our docs